src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/versionMatcher/versionMatcher.hpp

```diff
@@ -198,18 +203,19 @@ class VersionMatcher final
 
             if (*pVersionObjectA == *pVersionObjectB)
             {
-                return 0;
+                return VersionComparisonResult::A_EQUAL_B;
             }
             else if (*pVersionObjectA < *pVersionObjectB)
             {
-                return -1;
+                return VersionComparisonResult::A_LESS_THAN_B;
             }
             else
             {
-                return 1;
+                return VersionComparisonResult::A_GREATER_THAN_B;
             }
         }
-        return INVALID_VERSION_OBJECT_TYPE;
+
+        throw std::invalid_argument("Invalid version type");
```

Comment:
```suggestion
        logError(WM_VULNSCAN_LOGTAG, "Error comparing versions. Version objects are not valid.");        
```

I think we should avoid throwing an exception because this could stop the scan process. We can notify that there is an error, and continue the scanning.